feat: add conan datasource and manager (#12009)
renovatebot/renovate: lib/manager/conan/readme.md

> Extracts all conan packages in `conanfile.txt` or `conanfile.py`.

Reviewer: ```suggestion
Extracts all Conan packages in `conanfile.txt` or `conanfile.py`.
```

> Extracts all conan packages images in `conanfile.txt` or `conanfile.py`.

Reviewer 2: ```suggestion
Extracts all conan packages in `conanfile.txt` or `conanfile.py`.
```

Automated dependency updates for Conan

> Renovate supports upgrading dependencies in `conanfile.txt` or `conanfile.py` files.

Reviewer: ```suggestion
Renovate can upgrade dependencies in `conanfile.txt` or `conanfile.py` files.
```

How it works:

1. Renovate searches in each repository for any `conanfile.txt` or `conanfile.py` file
1. Renovate extracts existing dependencies from:
   - the `[requires]` and `[build_requires]` sections in the `conanfile.txt` format

>    - the `requirements()` and `build_requirements()` fucntions in the `conanfile.py` format

Reviewer 2: ```suggestion
   - the `requirements()` and `build_requirements()` functions in the `conanfile.py` format
```
